### PR TITLE
Plugins: Ensure catching all appropriate 4xx api/ds/query scenarios

### PR DESCRIPTION
**Pull request by wbrowne**
**What this PR does / why we need it**:

- Add missing catch for valid data source not found case (currently returns `500`, change to `404`)
- Add missing catch for plugin not actively registered (currently returns `500`, change to `404`)

### CODE REVIEW

File: pkg/api/metrics.go [hunk 1]
```diff
@@ -44,6 +34,38 @@ func (hs *HTTPServer) QueryMetricsV2(c *models.ReqContext) response.Response {
...
+	var badQuery *query.ErrBadQuery
+	if errors.As(err, &badQuery) {
+		return response.Error(http.StatusBadRequest, util.Capitalize(badQuery.Message), err)
+	}
+
+	if errors.Is(err, backendplugin.ErrPluginNotRegistered) {
+		return response.Error(http.StatusNotFound, "Plugin not found", err)
+	}
+
+	if errors.Is(err, backendplugin.ErrMethodNotImplemented) {
```
Comment (marefr) [inline pkg/api/metrics.go:53]: This feels like a plugin error and not grafana so maybe `502 Bad gateway` instead?

File: pkg/api/metrics.go [hunk 2]
```diff
@@ -44,6 +34,38 @@ func (hs *HTTPServer) QueryMetricsV2(c *models.ReqContext) response.Response {
...
+		return response.Error(http.StatusBadRequest, "Plugin functionality not implemented", err) // or 404
+	}
+
+	if errors.Is(err, backendplugin.ErrPluginUnavailable) {
+		return response.Error(http.StatusBadGateway, "Plugin unavailable", err)
+	}
+
+	var failedQuery *backendplugin.ErrFailedQuery
```
Comment (marefr) [inline pkg/api/metrics.go:61]: This could be 200/207 (if we decide to go that path) since an error originating from a plugin.

**Review by marefr — COMMENTED** [review]
As a side note. One case we're not handling today is timeouts, e.g. `504 Gateway Timeout`. Guess those ends up with a 500 currently.

**Review by papagian — COMMENTED** [review]
LGTM but if it's intentionally assigned to `8.5.2` milestone then needs to be back-ported.

**Comment by wbrowne** [comment]
Thanks for the reminder @papagian! 😄

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**